### ISSUE TITLE
`farmhash.h` error when building with `gnu++11` switch.

### ISSUE THREAD

**Issue by bartoszek** [OpenImageIO/oiio#2909]
I stumble on an issue with `oiio:2.2.12` when trying to build `alicevision` which has `c++11` compiler standard enabled.

After forcing `c++14` it builds fine.

Perhaps there should be a fail-safe warning about incompatible compiler standards?

`gcc:10.2.0 error message:`
```
/usr/include/OpenImageIO/detail/farmhash.h:1499:46: error: declaration of ‘__m128i Rotate32(__m128i, int)’ not in a namespace surrounding ‘OpenImageIO_v2_2::farmhash::inlined’
 1499 | STATIC_INLINE __m128i Rotate(__m128i x, int c) {
      |                                              ^
/usr/include/OpenImageIO/detail/farmhash.h:1499:46: error: ‘__m128i OpenImageIO_v2_2::farmhash::inlined::Rotate32(__m128i, int)’ should have been declared inside ‘OpenImageIO_v2_2::farmhash::inlined’
/usr/include/OpenImageIO/detail/farmhash.h: In function ‘__m128i OpenImageIO_v2_2::farmhash::inlined::Rotate32(__m128i, int)’:
/usr/include/OpenImageIO/detail/farmhash.h:1500:10: error: ‘Or’ was not declared in this scope; did you mean ‘OpenImageIO_v2_2::farmhash::farmhashsa::Or’?
 1500 |   return Or(_mm_slli_epi32(x, c),
      |          ^~
      |          OpenImageIO_v2_2::farmhash::farmhashsa::Or
/usr/include/OpenImageIO/detail/farmhash.h:1496:23: note: ‘OpenImageIO_v2_2::farmhash::farmhashsa::Or’ declared here
 1496 | STATIC_INLINE __m128i Or(__m128i x, __m128i y) { return _mm_or_si128(x, y); }
      |                       ^~
/usr/include/OpenImageIO/detail/farmhash.h: In function ‘uint32_t OpenImageIO_v2_2::farmhash::farmhashsa::Hash32(const char*, size_t)’:
/usr/include/OpenImageIO/detail/farmhash.h:1525:39: error: ‘Rotate32’ is not a member of ‘OpenImageIO_v2_2::farmhash’; did you mean ‘OpenImageIO_v2_2::farmhash::inlined::Rotate32’?
 1525 |     a = NAMESPACE_FOR_HASH_FUNCTIONS::Rotate32(a, 21);
      |                                       ^~~~~~~~
/usr/include/OpenImageIO/detail/farmhash.h:1499:23: note: ‘OpenImageIO_v2_2::farmhash::inlined::Rotate32’ declared here
 1499 | STATIC_INLINE __m128i Rotate(__m128i x, int c) {
```

https://github.com/alicevision/AliceVision/issues/1010
`env:{os:Linux,dist:Arch,gcc:10.2.0,oiio:2.2.12,cmake:3.19.7,ninja:1.10.2}`

**Comment by bartoszek** [comment]
Crude patch I've used:
```diff
diff --git a/src/cmake/Helpers.cmake b/src/cmake/Helpers.cmake
index 9c6f192a9..6a2084460 100644
--- a/src/cmake/Helpers.cmake
+++ b/src/cmake/Helpers.cmake
@@ -85,6 +85,7 @@ function(alicevision_add_library library_name)
     RUNTIME
         DESTINATION ${CMAKE_INSTALL_BINDIR}
   )
+    target_compile_features(${library_name} PUBLIC cxx_std_14)
 endfunction()
 
 # Add interface function
```

**Comment by lgritz** [comment]
Ah, so maybe it's gnu++11 that's the problem, but C++11 would be ok? Because our CI tests C++11 and has not produced these errors.

**Comment by bartoszek** [comment]
@lgritz Nope, just replaced `gnu++11` with `c++11` and same issue pops up.

**Comment by lgritz** [comment]
Strange. Our CI test that uses C++11 is using gcc 4.8. I wonder if for some reason this construct that was accepted for gcc4.8/C++11 is rejected by gcc10/C++11?

**Comment by lgritz** [comment]
Yes, I can repro this on our CI: it succeeds under gcc4.8/C++11 but fails with gcc10/C++11.
I will work on a fix shortly.

**Comment by bartoszek** [comment]
Looks that way, last working gcc on my system is `gcc:9.3.0` ( same problem affects `clang:11.1.1` )

**Comment by lgritz** [comment]
Proposed fix: #2914
